core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -113,10 +159,14 @@ public void writeMessage(InputStream message, int length, @Nullable Runnable acc
       framer.writePayload(message, length);
     }
 
-    // TODO(nmittler): add flow control.
     if (accepted != null) {
       accepted.run();
     }
+
+    // Indicate whether or not the transport is ready to send additional data. Calling this
+    // method has the side-effect of unsetting the shouldNotifyOnReady flag, which is fine since
+    // returning true here is equivalent to calling onReady anyway.
+    return shouldNotifyOnReady();
```

Comment:
This won't work like you expect, since the framer will basically never have written out the frame at this point. Instead, I guess flush() should return the boolean.